invisible_cities/io/mcinfo_io.py

```diff
@@ -172,13 +181,97 @@ def __call__(self, mctables: (tb.Table, tb.Table, tb.Table, tb.Table),
+                    orig_indx = tbl.file_index.unique()
+                    new_indx  = np.arange(first_indx                 ,
+                                          first_indx + len(orig_indx))
+                    tbl.file_index.replace(orig_indx, new_indx, inplace=True)
+                except AttributeError:
+                    tbl['file_index'] = first_indx
+            mcwriter_(df=tbl, table_name=key.name)
+    return write_mctables
+
+
+def check_last_merge_index(h5out : tb.file.File) -> int:
+    """
+    Get the last file index used to index merged files
+    """
+    if 'MC' in h5out.root:
+        if 'event_mapping' in h5out.root.MC:
+            return h5out.root.MC.event_mapping.cols.file_index[-1]
+    return 0
+
+
+def read_mc_tables(file_in : str                 ,
+                   evt_arr : np.ndarray          ,
+                   db_file : Optional[str] = None,
+                   run_no  : Optional[int] = None) -> Dict:
+    """
+    Reads all the MC tables present in
+    file_in and sorts the Dataframes for
+    each into a dictionary.
+    """
+    mctbls = get_mc_tbl_list(file_in)
+    if (len(evt_arr) > 0 and
+        len(np.intersect1d(evt_arr, get_event_numbers_in_file(file_in))) == 0):
+        raise IndexError
+    tbl_dict = {}
+    for tbl in mctbls:
+        if   tbl is MCTableType.hits                 :
+            hits = load_mchits_df(file_in).reset_index()
+            tbl_dict[tbl] = hits[hits.event_id.isin(evt_arr)]
+        elif tbl is MCTableType.particles            :
+            parts = load_mcparticles_df(file_in).reset_index()
+            tbl_dict[tbl] = parts[parts.event_id.isin(evt_arr)]
+        elif tbl is MCTableType.extents              :
+            pass
+        elif (tbl is MCTableType.sns_response or
+              tbl is MCTableType.waveforms      )    :
+            sns = load_mcsensor_response_df(file_in, return_raw=True)
+            tbl_key = MCTableType.sns_response
+            tbl_dict[tbl_key] = sns[sns.event_id.isin(evt_arr)]
+        elif (tbl is MCTableType.sensor_positions or
+              tbl is MCTableType.sns_positions      ):
+            pos = load_mcsensor_positions(file_in, db_file, run_no)
+            tbl_key = MCTableType.sns_positions
+            tbl_dict[tbl_key] = pos
+        elif tbl is MCTableType.generators           :
+            gen = load_mcgenerators(file_in)
+            tbl_dict[tbl] = gen[gen.evt_number.isin(evt_arr)]
+        elif tbl is MCTableType.configuration        :
+            config = load_mcconfiguration(file_in)
+            tbl_dict[tbl] = config
+        elif tbl is MCTableType.events               :
+            pass
+        elif tbl is MCTableType.event_mapping        :
+            evt_map = load_mcevent_mapping(file_in)
+            tbl_dict[tbl] = evt_map[evt_map.event_id.isin(evt_arr)]
+        else                                         :
+            raise TypeError("MC table has no reader")
+    return tbl_dict
```

Comment:
To what end?

We send an event list in most user cases and if not it will give you the information for all events and the complete event list can be got by calling the get_event_numbers_in_file function